src/uwtools/drivers/driver.py

```diff
@@ -282,11 +295,12 @@ def __init__(
         self,
         cycle: Optional[datetime] = None,
         leadtime: Optional[timedelta] = None,
-        config: Optional[Union[dict, Path]] = None,
+        config: Optional[Union[dict, str, YAMLConfig, Path]] = None,
         dry_run: bool = False,
         key_path: Optional[list[str]] = None,
         batch: bool = False,
         schema_file: Optional[Path] = None,
+        coupler: Optional[str] = None,
```

Comment:
Accept an optional `coupler` argument naming a block in the UW YAML config from which the `rundir:` and `execution:` values can be taken.